diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/SingleConstructorRule.kt

```diff
@@ -122,7 +125,15 @@ class SingleConstructorRule(configRules: List<RulesConfig>) : DiktatRule(
             }
             .distinct()
 
-        classNode.convertSecondaryConstructorToPrimary(secondaryCtor, declarationsAssignedInCtor, nonTrivialAssignments, otherStatements, comments)
+        // future init body
+        val expressions = (secondaryCtor.psi as KtSecondaryConstructor)
+            .bodyBlockExpression
+            ?.statements
+            ?.map { it.text }
+            ?.filter { expr -> expr in otherStatements.map { it.text } || expr in nonTrivialAssignments.keys.map { it.text } }
+            ?: emptyList()
+
+        classNode.convertSecondaryConstructorToPrimary(secondaryCtor, declarationsAssignedInCtor, nonTrivialAssignments, otherStatements, comments, expressions)
```

Comment:
I think it would be more clear to pass all statements into the function and then split them into groups there (if that's what you're doing and if it doesn't cause a lot of changes in existing code)

Reply:
Either I split those two functions into few small ones or I remain it as is. What is the best solution for now?

Reply:
Ok, let's keep it